elixirkit/lib/elixirkit/bundler.ex

```diff
@@ -0,0 +1,383 @@
+    launcher_path = "#{launcher_dir}/#{app_name}"
+    log(:green, "creating", launcher_path)
+    File.mkdir_p!(launcher_dir)
+    cache_dir = cache_dir()
+
+    targets = [
+      "macos-aarch64",
+      "macos-x86_64"
+    ]
+
+    for target <- targets do
+      otp_dir = "#{cache_dir}/otp-#{@otp_version}-#{target}"
+      cc_target = cc_target(target)
+
+      cmd("clang", [
+        "-c",
+        "-I",
+        "#{otp_dir}/usr/include",
+        "-target",
+        cc_target,
+        "-o",
+        "#{app_tmp}/ElixirKit.o.#{target}",
+        "#{@elixirkit_dir}/c_src/ElixirKit.m"
+      ])
+
+      cmd(
+        "swiftc",
+        [
+          "-parse-as-library",
+          "-import-objc-header",
+          "#{@elixirkit_dir}/c_src/ElixirKit.h",
+          "-lc++",
+          "-ltermcap",
+          "-target",
+          cc_target,
+          "-o",
+          "#{app_tmp}/launcher.#{target}",
+          "#{app_tmp}/Launcher.swift",
+          "#{@elixirkit_dir}/c_src/ElixirKit.swift",
+          "#{otp_dir}/usr/lib/liberl.a",
+          "#{app_tmp}/ElixirKit.o.#{target}"
+        ]
+      )
+
+      # this is a native runner, as opposed to simply a bash script, because the latter seems to
+      # be always executed using Rosetta in an universal app.
+      File.write!(
+        "#{app_tmp}/runner.swift",
+        EEx.eval_file("#{@elixirkit_dir}/c_src/runner.swift", release: release)
+      )
+
+      cmd(
+        "swiftc",
+        ~w(-o #{app_tmp}/runner.#{target} #{app_tmp}/runner.swift -target #{cc_target})
+      )
+    end
+
+    cmd(
+      "lipo",
+      Enum.map(targets, &"#{app_tmp}/launcher.#{&1}") ++ ~w(-create -output #{launcher_path})
+    )
+
+    cmd(
+      "lipo",
+      Enum.map(targets, &"#{app_tmp}/runner.#{&1}") ++ ~w(-create -output #{launcher_path}Runner)
+    )
+
+    erl_path = "#{app_dir}/Contents/Resources/rel/erts-#{release.erts_version}/bin/erl"
+
+    File.write!(erl_path, """
+    #!/bin/sh
+    SELF=$(readlink "$0" || true)
+    if [ -z "$SELF" ]; then SELF="$0"; fi
+    BINDIR="$(cd "$(dirname "$SELF")" && pwd -P)"
+    ROOTDIR="${ERL_ROOTDIR:-"$(dirname "$(dirname "$BINDIR")")"}"
+    EMU=beam
+    PROGNAME=$(echo "$0" | sed 's/.*\\///')
+    export EMU
+    export ROOTDIR
+    export BINDIR
+    export PROGNAME
+    RELEASE_ERLEXEC="${RELEASE_ERLEXEC:-"$BINDIR/erlexec"}"
+    exec $RELEASE_ERLEXEC ${1+"$@"}
+    """)
+
+    File.rm_rf!(app_tmp)
+
+    cmd(
+      "/System/Library/Frameworks/CoreServices.framework" <>
+        "/Versions/A/Frameworks/LaunchServices.framework" <>
+        "/Versions/A/Support/lsregister",
+      ["-f", app_dir]
+    )
+  end
+
+  defp cc_target("macos-aarch64"), do: "arm64-apple-macos11.0"
+  defp cc_target("macos-x86_64"), do: "x86_64-apple-macos11.0"
+
+  @doc false
+  def open_release_app do
+    %{name: name, dir: dir} = release_app_config()
+    ensure_not_running(name)
+    tty = tty()
+    forward_logs("#{System.user_home!()}/Library/Logs/#{name}.log")
```

Comment:
fyi: the runner spawns the launcher and redirects the IO to log file so here we're reading from it.